metaflow/datastore/datastore_set.py

```diff
@@ -30,7 +27,12 @@ def __init__(
             # produce a set of SHA keys to prefetch based on artifact names
             prefetch = set()
             for ds in self.task_datastores:
-                prefetch.update(ds.keys_for_artifacts(prefetch_data_artifacts))
+                l = [
```

Comment:
chage is because there can now be more than one key for each artifact so this is now a list of list.